plugins/modules/dcnm_image_policy.py

```diff
@@ -0,0 +1,694 @@
+    ImagePolicyCommon
+from ansible_collections.cisco.dcnm.plugins.module_utils.image_policy.create import \
+    ImagePolicyCreateBulk
+from ansible_collections.cisco.dcnm.plugins.module_utils.image_policy.delete import \
+    ImagePolicyDelete
+from ansible_collections.cisco.dcnm.plugins.module_utils.image_policy.endpoints import \
+    ApiEndpoints
+from ansible_collections.cisco.dcnm.plugins.module_utils.image_policy.image_policies import \
+    ImagePolicies
+from ansible_collections.cisco.dcnm.plugins.module_utils.image_policy.image_policy_task_result import \
+    ImagePolicyTaskResult
+from ansible_collections.cisco.dcnm.plugins.module_utils.image_policy.params_spec import \
+    ParamsSpec
+from ansible_collections.cisco.dcnm.plugins.module_utils.image_policy.payload import \
+    Config2Payload
+from ansible_collections.cisco.dcnm.plugins.module_utils.image_policy.query import \
+    ImagePolicyQuery
+from ansible_collections.cisco.dcnm.plugins.module_utils.image_policy.replace import \
+    ImagePolicyReplaceBulk
+from ansible_collections.cisco.dcnm.plugins.module_utils.image_policy.update import \
+    ImagePolicyUpdateBulk
+
+
+def json_pretty(msg):
+    """
+    Return a pretty-printed JSON string for logging messages
+    """
+    return json.dumps(msg, indent=4, sort_keys=True)
+
+
+class ImagePolicyTask(ImagePolicyCommon):
+    """
+    Create, delete, query, replace, or update image policies
+    """
+
+    def __init__(self, ansible_module):
+        super().__init__(ansible_module)
+        self.class_name = self.__class__.__name__
+        method_name = inspect.stack()[0][3]
+
+        self.log = logging.getLogger(f"dcnm.{self.class_name}")
+        self.log.debug("ENTERED ImagePolicyTask()")
+
+        self.endpoints = ApiEndpoints()
+        self.have = None
+        self.idempotent_want = None
+
+        self.path = None
+        self.verb = None
+
+        self.config = ansible_module.params.get("config", {})
+        self._valid_states = ["deleted", "merged", "overridden", "query", "replaced"]
+
+        if not isinstance(self.config, list):
+            msg = f"{self.class_name}.{method_name}: "
+            msg += "expected list type for self.config. "
+            msg += f"got {type(self.config).__name__}"
+            self.ansible_module.fail_json(msg)
+
+        self.want = []
+        self.need = []
+        # policies to created
+        self.need_create = []
+        # policies to updated
+        self.need_delete = []
+        # policies to deleted
+        self.need_update = []
+        # policies to query
+        self.need_query = []
+        self.validated_configs = []
+
+        self.task_result = ImagePolicyTaskResult(self.ansible_module)
+        self.task_result.changed = False
+
+    def get_have(self) -> None:
+        """
+        Caller: main()
+
+        self.have consists of the current image policies on the controller
+        """
+        self.log.debug("ENTERED")
```

Comment:
Did you want to commit these log messages or do they need to be removed before merging?

Reply:
They're pretty helpful when debugging issues and won't be logged if logging is disable (the default).  If OK, can we leave them?  Else I can remove them (I'm not strongly-partial either way).